### PR TITLE
Set requested music as current only if it is actually played

### LINKED ISSUE #1527: MIDI music restarts after checking puzzle map and making any movement
When hero checks puzzle mal, and exit puzzle map window music is still playing without interruption.
But after hero makes any move the composition starts playing from the beginning.

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #1527

### CODE REVIEW

File: src/fheroes2/kingdom/puzzle.cpp [hunk 1]
```diff
@@ -90,9 +90,6 @@ void Puzzle::ShowMapsDialog( void ) const
     const fheroes2::Display & display = fheroes2::Display::instance();
     int old_cursor = cursor.Themes();
 
-    if ( !Settings::Get().MusicMIDI() )
-        AGG::PlayMusic( MUS::PUZZLE );
```
Comment (ihhub) [inline src/fheroes2/kingdom/puzzle.cpp:94]: External music has this song while MIDI doesn't. Does the game work without this change?

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left just one comment. Would you mind to take a look?

**Comment by oleg-derevenetz** [comment]
Hi @ihhub

> External music has this song while MIDI doesn't. Does the game work without this change?

Yes it does, but `PlayMusic()` with the same music is called later inside `if ( !sf.empty() )` body, so one of these calls is redundant. With proposed changes in `PlayMusic()` there is no need in this additional check anyway, because there will be no side effects. The only thing - does this MUZ::PUZZLE require looping or not? Those two `PlayMusic()` calls are contradictory in this regard.

**Comment by ihhub** [comment]
> Hi @ihhub
> 
> > External music has this song while MIDI doesn't. Does the game work without this change?
> 
> Yes it does, but `PlayMusic()` with the same music is called later inside `if ( !sf.empty() )` body, so one of these calls is redundant. With proposed changes in `PlayMusic()` there is no need in this additional check anyway, because there will be no side effects. The only thing - does this `MUZ::PUZZLE` require looping or not? Those two `PlayMusic()` calls are contradictory in this regard.
> 
> P.S. It looks like in the OG `MUZ::PUZZLE` is played only once, so the second call is more relevant.

Ah, I missed another place below as GitHub wasn't showing all the code within the function.

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you very much for the fix!

**Pull request merged**